src/proto/connection.rs

```diff
@@ -145,7 +145,9 @@ where
 
     /// connection flow control
     pub(crate) fn set_target_window_size(&mut self, size: WindowSize) {
-        self.inner.streams.set_target_connection_window_size(size);
+        let _res = self.inner.streams.set_target_connection_window_size(size);
+        // TODO: proper error handling
+        debug_assert!(_res.is_ok());
```

Comment:
There are some debug left around here and in other files. Are these on purpose?

Reply:
Yes. Whenever it was obvious on how to handle a potential overflow, I did. However, many of the call sites do not have a `Result` return type, so it is not easy to propagate the Error. I assume that those functions should not be able not fail. I did not want to incur additional overhead in release builds, so I stuck with `debug_assert!` and left the `TODO` comment. For those TODOs someone with more experience with the code should check whether they are fine to be left as `debug_assert!`, whether they should be checked with `assert!` or whether error propagation is needed.

Reply:
FYI: I did more fuzzing runs with this patch and none of the `debug_assert!` are triggered, so that gives some assurance that the they are indeed not reachable. So I think we could remove the todo comments, but I would leave the `debug_assert!` as is.